py/obj.h

```diff
@@ -668,6 +668,11 @@ extern const mp_obj_type_t mp_type_stringio;
 extern const mp_obj_type_t mp_type_bytesio;
 extern const mp_obj_type_t mp_type_reversed;
 extern const mp_obj_type_t mp_type_polymorph_iter;
+#if MICROPY_ENABLE_FINALISER
+extern const mp_obj_type_t mp_type_polymorph_iter_with_finaliser;
+#else
+#define mp_type_polymorph_iter_with_finaliser mp_type_polymorph_iter
+#endif
```

Comment:
Unless `MICROPY_ENABLE_FINALISER` is enabled, it shouldn't be possible to use `mp_type_polymorph_iter_with_finaliser` (and so remove the else).